### DISCUSSION TITLE
General Discussion for Questions and Comments

### DISCUSSION THREAD

**Discussion by lageist** [UNC-Project-Open-AAC/OS-DPI#87] [General]
Post or respond to any topics or questions of general interest here.

**Comment by jbizovi** [comment]
@gbishop I'm hoping you can answer some of my questions below based on the attached docs.
My Goal: Take the Contact osdpi and xlsx and "simplify" them to reflect a single screen with just the Chat (greeting and meeting) tab and Quick Grid on the right. Instead of building up from an empty layout and spreadsheet, I instead tried to take things away to simplify what already existed.

My Reality: It worked temporarily and I had a working display with all additional tabs/content removed. Then, when I reloaded the spreadsheet in an incognito tab, the text in the main grid would not load. I'm not sure what caused it, since I couldn't tell what changed. Now, it will not work for me anymore.

1. I tried to upload my osdpi file here, but that type of file isn't supported, so I guess my first question is how do I upload that here? In the meantime, I've uploaded all relevant docs to UNC-Project-Open-AAC/Internal/Jenna
2. I'm not sure what caused my display to work one time and then not another when supposedly the content spreadsheet was the same both times. Could it be because I was trying to access it in 2 places?
3. I notice that when I work in the OSDPI that after some time, when I go to content, the local doc I've uploaded no longer shows there even though content is clearly still being pulled from it. Why does this happen? Does it time out? Am I just not seeing something?
4. I tried out a few different positions for the who/what buttons. Is there a way to have them on the left stacked one on top of the other to fill that portion of the screen?

**Comment by gbishop** [comment]
1. Yes that is a stupid misfeature of github issues/disucssions (IMHO).
Instead you need to go here:
https://github.com/UNC-Project-Open-AAC/OS-DPI/tree/main/Designs%20In%20Progress
and click the Add File button and choose upload file. Then, you'll be able
to upload it and give me a link to that.
2. I'm guessing some state variable was defined by the original version
that you are depending on. Since you started with the original, the
variable had the correct value, now it doesn't. I suggest you open them
side by side. Your version running after the old one and your version
loaded anew. Look at the variables in the lower left.
3. If you reload the page, it will lose the connection of the file
you previously loaded. Other than that it may be a bug.
4. I *think* if you put the radio group in a stack and adjust the scale it
should do what you want. But now that I have tried it, it doesn't work. I'm
forcing the button width so you get at least 2 on a line. That is a hack
that got the job done at the moment. You should be able to set the "scale"
or something on the buttons to make them line up like you want. File an
issue please.

I looked at your design.

You don't really need to use the #topic = $ChatTopic since you only have 1.
I would drop that filter completely. Also on the radio.
When I changed that it started working.

You don't need the tab control though I see why you kept it because we
don't allow you to replace top level stuff. Copy and Paste should fix that
later.

**Reply by jbizovi** [reply]
Looking into all of this now. I don't think I'm able to add documents in the location you shared, however. Below is a screen shot of the message I see when I try to do that.
<img width="1262" alt="Untitled" src="https://user-images.githubusercontent.com/97691300/156386002-9736dd8f-042b-4b1b-b75a-6348020b6782.png">

**Comment by gbishop** [comment]
Ah, we can fix that. You have been elevated. Use your newfound power with
care.

**Reply by jbizovi** [reply]
Okay, I've tried everything based on your comments and am stuck. Are you free for a zoom this afternoon to explore your responses to numbers 2 and 4? Thanks!

**Reply by jbizovi** [reply]
Looks like I'm still unable to upload documents to OS-DPI public. That's where I'd like to put the video recording of our session we just completed.
<img width="1351" alt="Screen Shot 2022-03-02 at 2 33 25 PM" src="https://user-images.githubusercontent.com/97691300/156435881-329991b0-2944-4c98-84c9-f7d9da79e6a2.png">

**Comment by jbizovi** [comment]
Learnings from our zoom call which I will post as soon as I can:

- $ChatTopic was still showing up under the $ChatAspect radio heading and Grid. Once we deleted this, everything worked again.
- Hitting the "Clear State" button after changes will help refresh sooner to identify when/where changes break the interface
- Duplicating tabs will change the interface in both places. Instead, export the osdpi file, open a new tab, and import. The osdpi will autosave under an indexed name.
- All slot actions could be deleted.
- Since we deleted slots, I could change the spreadsheet from $$Identifiers=name$$ to just name.
- There is no easy way right now to copy and paste or move things up in a hierarchical sense. Gary is thinking through this. In the meantime, I can delete TabControl and move all other tiers "up" manually.

**Reply by jbizovi** [reply]
Contact documents are located [here](https://github.com/UNC-Project-Open-AAC/OS-DPI/tree/main/Designs%20In%20Progress/Contact%20Simplified). The video file was too large to upload. I'm happy to share with anyone interested in watching.

**Comment by gbishop** [comment]
They could be **much** smaller. If I get a chance I'll see if I can find a
tool we could use to compress them.
